pinot-spi/pom.xml

```diff
@@ -33,6 +33,7 @@
   <url>https://pinot.apache.org/</url>
   <properties>
     <pinot.root>${basedir}/..</pinot.root>
+    <shade.phase.prop>package</shade.phase.prop>
```

Comment:
Seems it is not enabled by default. Do you enable it by explicitly setting the flag?

Reply:
Yes. It used to be enabled by the shading profile. Now that we have removed those we should have this set. I followed the change pattern in pom of pinot-common in https://github.com/apache/pinot/pull/12712

Reply:
changed to disabled by default and tested with mvn build, can you take another look @Jackie-Jiang